### PR TITLE
PMM-7628 GitHub actions migration

### PR DESCRIPTION
**Pull request by lunaticusgreen**

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+            ~/.npm
+          key: ${{ runner.os }}-node-${{ matrix.node-version }}-yarn-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-${{ matrix.node-version }}-yarn-
+
+      - run: make prepare_release
+      - run: make generate_coverage
+
+  workflow_success:
+    if: ${{ success() }}
```
Comment (yurkovychv) [inline .github/workflows/build.yml:124]: this line is not needed here,`job.needs` [automatically waits](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idneeds) for the outcome of a successful job

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [12.x]
+
+    steps:
+      - uses: actions/checkout@v2
```
Comment (denisok) [inline .github/workflows/build.yml:14]: according to @atymchuk all actions should be from percona-platform, probably for the security. If that works now we need to restrict it in settings.

If there missing actions, we could fork it.

**Comment by denisok** [comment]
@atymchuk should we restrict Actions to percona-platform on all repos ?

File: .github/workflows/build.yml (hunk 1 above)
Comment (lunaticusgreen) [reply .github/workflows/build.yml:120]: So it will be just ignored if any of the dependencies will fail?

File: .github/workflows/build.yml (hunk 2 above)
Comment (lunaticusgreen) [reply .github/workflows/build.yml:14]: @denisok Thanks, fixed everywhere

File: .github/workflows/build.yml (hunk 1 above)
Comment (yurkovychv) [reply .github/workflows/build.yml]: according to documentation yes

**Comment by lunaticusgreen** [comment]
@denisok @atymchuk Replaced actions with percona-platform for all actions

File: .github/workflows/build.yml (hunk 1 above)
Comment (lunaticusgreen) [reply .github/workflows/build.yml]: Nice, thanks, fixed

**Review by yurkovychv — APPROVED** [review]

**Review by denisok — APPROVED** [review]

**Review by atymchuk — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+          key: ${{ runner.os }}-node-${{ matrix.node-version }}-yarn-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-${{ matrix.node-version }}-yarn-
+
+      - run: make prepare_release
+      - run: make docker_clean
+      - run: make e2e_local
+
+      - name: Publish Test Report
+        uses: mikepenz/action-junit-report@v2
```
[the remark this answers is not in the document]
Comment (lunaticusgreen) [reply .github/workflows/build.yml:77]: Discussed it, Alex will fork this action to our namespace and merge this PR after that

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [12.x]
+
+    steps:
+      - uses: percona-platform/checkout@v2
+      - name: Build grafana-dashboards
+        uses: percona-platform/setup-node@v2
```
Comment (puneet0191) [inline .github/workflows/build.yml:16]: This is using node 12, please change it

**Review by puneet0191 — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml (hunk 4 above)
Comment (lunaticusgreen) [reply .github/workflows/build.yml:16]: Done, thanks for pointing it!

File: .github/workflows/build.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (atymchuk) [reply .github/workflows/build.yml]: Forked! Now I'm migrating saas-ui to make tests pass :)

**Review by atymchuk — APPROVED** [review]

**Pull request merged**